Source/GOATracer/ViewModels/MainWindowViewModel.cs

```diff
@@ -0,0 +1,214 @@
+        /// </summary>
+        private void AddNewLight()
+        {
+            _lightCounter++;
+            var newLightModel = new Light() { name = $"Light {_lightCounter}", isEnabled = true };
+            _rayTracerModel.Lights.Add(newLightModel);
+
+            var newLightVM = new LightViewModel(newLightModel, DeleteLight);
+            newLightVM.PropertyChanged += LightViewModel_PropertyChanged;
+            Lights.Add(newLightVM);
+            UpdateEnabledLights();
+        }
+
+        /// <summary>
+        /// Updates the collection of enabled lights by clearing the current list and adding all lights that are
+        /// enabled. This ensures that the EnabledLights collection always reflects the current state of the Lights collection.
+        /// Also important to keep the enabled lights in sync for the UI and rendering
+        /// </summary>
+        private void UpdateEnabledLights()
+        {
+            EnabledLights.Clear();
+            foreach (var l in Lights.Where(l => l.IsEnabled))
+                EnabledLights.Add(l);
+        }
+
+        // Event handler for property changes in LightViewModel
+        private void LightViewModel_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(LightViewModel.IsEnabled))
+                UpdateEnabledLights();
+        }
+
+
+        /// <summary>
+        /// Executes the specified command based on the provided action string.
+        /// </summary>
+        /// <param name="action">A string representing the command to execute. Supported values are: 
+        /// "Render": Opens a new render window.
+        /// "DeleteLight": Deletes the currently selected light, if one is selected.
+        /// "AddLight": Adds a new light to the scene.</param>
+        public void OnCommand(string action)
+        {
+            switch (action)
+            {
+                case "Render":
+                    var renderWindow = new RenderWindow();
+                    renderWindow.Show();
+                    break;
+                case "AddLight":
+                    AddNewLight();
+                    break;
+
+                default:
+                    break;
+            }
+
+
+        }
+
+        /// <summary>
+        /// Deletes a light with the specified name from the collection of lights in both the ViewModel and the Model.
+        /// Deletion executed by name, not by reference.
+        /// </summary>
+        /// <param name="lightName">The name of the light to delete</param>
+        private void DeleteLight(String lightName)
+        {
+            var lightVM = Lights.FirstOrDefault(l => l.Name == lightName);
+            if (lightVM != null)
+            {
+                bool wasSelected = SelectedLight == lightVM;
+                Lights.Remove(lightVM);
+                _rayTracerModel.Lights.Remove(lightVM.Model);
```

Comment:
```suggestion
                RayTracerModel.Lights.Remove(lightVM.Model);
```
Reference the property instead of the backing field.